python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -156,6 +156,18 @@ def core_execute_run(
         raise
 
 
+def _instance_from_ref(instance_ref: InstanceRef, partitions_def: PartitionsDefinition):
+    # Certain gRPC servers do not have access to the instance, so we only attempt to instantiate
```

Comment:
- A helper method here is reasonable, but let's call it `_instance_from_ref_for_dynamic_partitions` to communicate that it's not always yielding an instance.
- This needs a `@contextmanager` decorator and the result yielded instead of returned to work as intended
- Annotate return vale as `Iterator[Optional[DagsterInstance]]`
- This should never be called with a `partitions_def=None`, so no need to check

Reply:
thanks, applied these changes in last commit